src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -241,26 +190,18 @@ class TaskReplaceActor(
     }
   }
 
-  def checkFinished(): Unit = {
-    if (targetCountReached(runSpec.instances) && oldActiveInstanceIds.isEmpty) {
+  def checkFinished(nr_new: Int, nr_old: Int): Unit = {
```

Comment:
variable names should say they explicitely wait for new instances in healthy state.

Reply:
Actually, we don't check the last instances are Healthy. To me, it's enough to say a deployment is finished when all instances run on the last runSpec.

Reply:
I don't think that's the current behavior of marathon. For instance, if you have an app with 1 instance with a minimalHealthCapacty at 0 and maximumOverCapacity at 0, the deployment is over once the new instance is healthy, not whne it is just started.

Reply:
I was wrong I indeed took into account only healthy new instances when I wrote the patch. I just renammed variables and added a comment to make it clearer.